src/index.ts

```diff
@@ -1 +1,254 @@
-console.log('Hello');
+import * as fs from 'fs';
+import * as path from 'path';
+import * as yargs from 'yargs';
+
+enum TypeEnum {
+  Function,
+  Object,
+  Error,
+}
+
+class TypeTree {
```

Comment:
Could we split this into the following classes?:

- `class BaseType`
- `class FunctionType extends BaseType`
- `class ObjectType extends BaseType`
- `class ErrorType extends BaseType`